### ISSUE TITLE
Detecting Pyodide, at build time

### ISSUE THREAD

**Issue by gabrielfougeron** [pyodide/pyodide#3493]
## 📚 Documentation

The current documentation reads:

To detect Pyodide, at build time use,

```
import os

if "PYODIDE" in os.environ:
    # building for Pyodide
```

This used to work on my computer using pyodide-build v0.21.

When I tried it using pyodide-build v0.22 however, the test `if "PYODIDE" in os.environ` returned false.

Inspecting my `os.environ`, I changed the test to `if ("PYODIDE_ROOT" in os.environ):`, and my setup.py detects when the code is being built with pyodide.

This lead me to believe the docs needed updating. However, searching the pyodide repo, I saw `if "PYODIDE" in os.environ` in the following patch file: https://github.com/pyodide/pyodide/blob/main/packages/Pillow/patches/0001-Enable-image-formats.patch

So I'm not really sure anymore ...

**Comment by rth** [comment]
For in-tree build the PYODIDE env variable is still [set here](https://github.com/pyodide/pyodide/blob/main/Makefile.envs#L45) and also set in [get_bash_runner](https://github.com/pyodide/pyodide/blob/main/pyodide-build/pyodide_build/buildpkg.py#L120), so that should work.

For out of tree builds with `pyodide build` it  does indeed looks like it's not set . The Makefile.env shipped in   `xbuildenv-0.22.0.tar.bz2 ` is never loaded, unless I'm missing something.  Maybe @hoodmane could confirm.

 If that's true, we could add it [here](https://github.com/pyodide/pyodide/blob/3c92574c57e8a57d5e2b081e9a19b01cfc9b3745/pyodide-build/pyodide_build/out_of_tree/build.py#L19) I think:
```py
env_vars = os.environ.copy()
env_vars['PYODIDE'] = 1

build_env_ctx = pypabuild.get_build_env(
        ...,
        env=env_vars
)
```

**Comment by hoodmane** [comment]
I think we should extract these lines:
https://github.com/pyodide/pyodide/blob/main/pyodide-build/pyodide_build/buildpkg.py#L119-L133
into a separate function in `common.py`. Or at least whatever part of it should be shared with `pyodide build`.
